src/Agent.Listener/Agent.cs

```diff
@@ -120,22 +120,12 @@ public override void Initialize(IHostContext hostContext)
 
                 AgentSettings settings = configManager.LoadSettings();
                 bool runAsService = configManager.IsServiceConfigured();
-                if (command.Run || !runAsService)
-                {
-                    // Run the agent interactively
-                    Trace.Verbose($"Run as service: '{runAsService}'");
-                    return await RunAsync(TokenSource.Token, settings, runAsService);
-                }
-
                 if (runAsService)
                 {
-                    // This is helpful if the user tries to start the agent.listener which is already configured or running as service
-                    // However user can execute the agent by calling the run command
-                    // TODO: Should we check if the service is running and prompt user to start the service if its not already running?
-                    _term.WriteLine(StringUtil.Loc("ConfiguredAsRunAsService"));
+                    Trace.Info($"Agent is configured to run as service.");
```

Comment:
We should break this so that `run` does not walk through config.

One of the things I don't like about the experience is, we don't validate if you pass an incorrect parameter (e.g. typo "--srverurl ..."). Having run implicitly call config complicates this.

Also it would be nice to break usage into separate commands, rather than one giant usage string that covers all subcommands.